src/utils.rs

```diff
@@ -82,84 +82,73 @@ pub fn find_message_type_in_file_descriptor(
       message_name, descriptor.name.as_deref().unwrap_or("unknown")))
 }
 
-/// Search for a message by type name in the file descriptor, and if not found, search in all the
-/// descriptors
-pub fn find_message_type_in_file_descriptors(
-  message_type: &str,
-  file_descriptor: &FileDescriptorProto,
-  all_descriptors: &HashMap<String, &FileDescriptorProto>
-) -> anyhow::Result<DescriptorProto> {
-  find_message_type_in_file_descriptor(message_type, file_descriptor)
-    .or_else(|_| {
-      all_descriptors.values()
-        .find_map(|fd| find_message_type_in_file_descriptor(message_type, fd).ok())
-        .ok_or_else(|| anyhow!("Did not find a message type '{}' in any of the file descriptors", message_type))
-    })
-}
-
-// Search for a message by name, first in the current file descriptor, then in all descriptors.
+/// Very similar to find_message_type_by_name, but it narrows down a list of file descriptors to the ones where package
+/// name matches the given package name, either the one provided explicitly, or the one from the file_descriptor.
 pub fn find_message_descriptor(
-  message_name: &str,
-  package: Option<&str>,
-  file_descriptor: &FileDescriptorProto,
-  all_descriptors: &HashMap<String, &FileDescriptorProto>
+    message_name: &str,
+    package: Option<&str>,
+    file_descriptor: &FileDescriptorProto,
+    all_descriptors: &HashMap<String, &FileDescriptorProto>,
 ) -> anyhow::Result<DescriptorProto> {
-  if let Some(package) = package {
-    find_message_with_package_in_file_descriptors(message_name, package, file_descriptor, all_descriptors)
-  } else {
-    find_message_type_in_file_descriptors(message_name, file_descriptor, all_descriptors)
-  }
+    let package = package
+        .filter(|p| !p.is_empty())
+        .unwrap_or(file_descriptor.package.as_deref().unwrap_or_default());
+    debug!(
+        "Looking for message '{}' in package '{}'",
+        message_name, package
+    );
+    find_all_file_descriptors_for_package(package, all_descriptors)
+        .unwrap()
```

Comment:
You should avoid using unwraps, as they will case a panic if there is no value. Panics aren't always handled like exceptions, some environments (like embedded) they result in process aborts.

Unwraps are ok to use if you know for certain they contain a value or they are used in a test, as that will just cause the test to fail if there is no value.

Better to use the `?` operator (as the function returns a result). This will cause it to bail from the function if it is an error, otherwise just unwrap the value.

Reply:
done